checkov/common/bridgecrew/integration_features/features/policy_metadata_integration.py

```diff
@@ -157,6 +164,31 @@ def _handle_customer_prisma_policy_metadata(self, prisma_policy_metadata: list[d
                     ckv_id = self.get_ckv_id_from_pc_id(pc_id)
                     if ckv_id:
                         self.filtered_policy_ids.append(ckv_id)
+            self._add_ckv_id_for_filtered_cloned_checks()
+
+    def _add_ckv_id_for_filtered_cloned_checks(self) -> None:
+        """
+        Filtered checks are the policies that are returned by --policy-metadata-filter.
+        Cloned checks are policies that have modified metadata in Prisma (severity, title etc).
+        Filtered checks do not have a definition if they are cloned, instead they have a sourceIncidentId
+        which corresponds to the BC ID of the original source check.
+        This method adds the CKV ID for that source check to the list of filtered policies to ensure it is run.
+        Example:
+            Input:
+                filtered_policy_ids = [ "org_AWS_1609123441" ]
+                ckv_id_to_source_incident_id_mapping =  { "org_AWS_1609123441": "BC__AWS_GENERAL_123" }
+                bc_id_to_ckv_id_mapping = { "BC__AWS_GENERAL_123": "CKV_AWS_123" }
+            Output:
+                filtered_policy_ids = [ "org_AWS_1609123441", "CKV_AWS_123" ]
+        """
+        ckv_ids = []
+        for policy_id in self.filtered_policy_ids:
+            source_bc_id = self.get_source_incident_id_from_ckv_id(policy_id)
+            if source_bc_id:
+                ckv_id = self.get_ckv_id_from_bc_id(source_bc_id)
+                if ckv_id:
+                    ckv_ids.append(ckv_id)
```

Comment:
@kartikp10 Just a general engineering tip - try to avoid this nested if pattern by using the `fail-fast` pattern. Might be a little longer but needs less context to understand:
```suggestion
            if not source_bc_id:
                continue
            ckv_id = self.get_ckv_id_from_bc_id(source_bc_id)
            if not ckv_id:
                continue
            ckv_ids.append(ckv_id)
```